src/main/java/net/neoforged/neoforge/client/extensions/IBlockModelGeneratorExtensions.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.extensions;
+
+import net.minecraft.client.data.models.BlockModelGenerators;
+import net.minecraft.client.data.models.blockstates.MultiVariantGenerator;
+import net.minecraft.client.data.models.model.ModelLocationUtils;
+import net.minecraft.client.data.models.model.ModelTemplates;
+import net.minecraft.client.data.models.model.TextureMapping;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+
+public interface IBlockModelGeneratorExtensions {
```

Comment:
All methods in this class should have a short doc comment mentioning that they are standalone versions of the respective methods in `BlockStateGenerators.BlockFamilyProvider`.
The interface's name should also match the target class's name exactly to avoid confusion.